src/components/DropDownButton.tsx

```diff
@@ -133,6 +135,22 @@ class DropDownButton extends React.Component<
 		};
 	}
 
+	componentDidMount() {
```

Comment:
So according to this https://reactjs.org/docs/react-component.html#setstate

> `setState()` will always lead to a re-render unless shouldComponentUpdate() returns false. 

And this handler is mounted even for closed dropdowns, if I'm reading it correctly. So if I have 10 closed dropdowns on page, click on something unrelated, it will cause 10 re-renders?

Reply:
@JackuB You are right, it will cause re-renders. Not a big issue in this case, but I agree that it can be improved by attaching the handlers when a component opens, and removing them when it closes.